rasa/utils/tensorflow/data_generator.py

```diff
@@ -0,0 +1,116 @@
+from typing import List, Union, Text, Optional, Any, Tuple
+
+import numpy as np
+import tensorflow as tf
+
+from rasa.utils.tensorflow.constants import SEQUENCE, BALANCED
+from rasa.utils.tensorflow.model_data import RasaModelData
+
+
+class IncreasingBatchSizeDataGenerator(tf.keras.utils.Sequence):
```

Comment:
is it tight in to `RasaModelData`? Does it make sense to move generator related methods fully to simplify `RasaModelData` class?

Reply:
You mean removing the method `as_tf_dataset` and everything related to that? We could. We could also add a `FixBatchSizeGenerator` that we could use during prediction so that we don't rely on `tf.data.Dataset.from_generator` anymore.

Reply:
yes, because I think `RasaModel` contains so many things, it became incomprehensible.
Let's remove `tf.data.Dataset.from_generator` isn't it deprecated?

Reply:
Already on it. Becomes much cleaner 🎉 
`tf.data.Dataset.from_generator` is not deprecated but has its limitations.